src/pipeline.ts

```diff
@@ -335,28 +339,108 @@ export async function checkNewCandidateMatch(
 			label: searcheeLabel,
 			message: `Did not find an existing entry using ${method} for ${candidateLog}`,
 		});
-		return { decision: null, actionResult: null };
+		return null;
 	}
-	const searchees: SearcheeWithLabel[] = filterDupesFromSimilar(
-		metas
-			.map(createSearcheeFromMetafile)
-			.filter(isOk)
-			.map((r) => r.unwrap())
-			.map((searchee) => ({ ...searchee, label: searcheeLabel }))
-			.filter((searchee) => filterByContent(searchee)),
+	searchees.push(
+		...filterDupesFromSimilar(
+			metas
+				.map(createSearcheeFromMetafile)
+				.filter(isOk)
+				.map((r) => r.unwrap())
+				.map((searchee) => ({ ...searchee, label: searcheeLabel }))
+				.filter((searchee) => filterByContent(searchee)),
+		),
 	);
 	if (!searchees.length) {
 		logger.verbose({
 			label: searcheeLabel,
 			message: `No valid entries found using ${method} for ${candidateLog}`,
 		});
-		return { decision: null, actionResult: null };
+		return null;
+	}
+	return method;
+}
+
+async function pushEnsembleForCandidate(
+	candidate: Candidate,
+	searchees: SearcheeWithLabel[],
+	searcheeLabel: SearcheeLabel,
+): Promise<string | null> {
+	const { seasonFromEpisodes } = getRuntimeConfig();
+	if (!seasonFromEpisodes) return null;
+	const seasonKey = getSeasonKey(stripExtension(candidate.name));
+	if (!seasonKey) return null;
+
+	const candidateLog = `${chalk.bold.white(candidate.name)} from ${candidate.tracker}`;
+	const { ensembleTitle, keyTitle, season } = seasonKey;
+	const key = `${keyTitle}.${season}`;
+	const ensemble = await memDB("ensemble").where({ ensemble: key });
+	if (ensemble.length === 0) {
+		logger.verbose({
+			label: searcheeLabel,
+			message: `Did not find an ensemble ${ensembleTitle} for ${candidateLog}`,
+		});
+		return null;
+	}
+	const files = ensemble
+		.map((e) => ({
+			path: e.absolute_path,
+			name: e.name,
+			length: e.length,
+		}))
+		.filter((f) => fs.existsSync(f.path));
+	if (files.length === 0) {
+		logger.verbose({
+			label: searcheeLabel,
+			message: `Did not find any files for ensemble ${ensembleTitle} for ${candidateLog}: sources may be incomplete or missing`,
+		});
+		return null;
 	}
+	// Get searchee.length by total size of each episode (average if multiple files for episode)
+	const uniqueElements = new Set(ensemble.map((e) => e.element));
+	const totalLength = [...uniqueElements].reduce((acc, cur) => {
```

Comment:
should we even bother with length checks for ensemble searchees? guessing the number of searchees feels like an easy way to get false negatives. I guess a lot of the time you'll have all of the episode files but you might also have all but one episode, and the finale might be twice as long. Would source/resolution and release group be enough?

Reply:
I think it's still helpful to reduce snatches. Since we are checking source, resolution, and release group, the only differences between episodes would be a couple bytes (so corrupted cross seeds or repacks). So the total size should still be close if the user has all.

If they are missing some, it's tricky but the lower limit for `seasonFromEpisodes` is `0.5` so there it should get caught. Different episode lengths and other things can reduce this effectiveness which can only be solved by sonarr. But I haven't seen any problems manifest and can always tweak it later on.